### DISCUSSION TITLE
官方文档有太多空缺，好多配置项都没有用法和参数解释

### DISCUSSION THREAD

**Discussion by oushu1suzhijie1** [antvis/G2#4977] [Q&A]

**Comment by pearmini** [comment]
嗯嗯，文档会在这两个月逐步完善

**Comment by wilson921108** [comment]
文档查的挺痛苦,ts类型声明也很痛苦.

**Comment by pearmini** [comment]
> 文档查的挺痛苦,ts类型声明也很痛苦.

文档的搜索功能正在添加！TS 声明是太多 any 了吗？

**Comment by wilson921108** [comment]
any还是没用到过, 不过string有碰到, 比如 

export declare type ColorAttr = string | string[] | ((datum: Datum, defaultColor?: string) => string);

实际使用的时候其实除了 string 还有 另一种声明颜色的方式
"l(90) 0:#ffffff 1:green 0.15"
这种算是渐变色. 但是被包含在string 里面.
同理 any 应该也有类似的问题, 而且更严重

**Comment by pearmini** [comment]
> any还是没用到过, 不过string有碰到, 比如
> 
> export declare type ColorAttr = string | string[] | ((datum: Datum, defaultColor?: string) => string);
> 
> 实际使用的时候其实除了 string 还有 另一种声明颜色的方式 "l(90) 0:#ffffff 1:green 0.15" 这种算是渐变色. 但是被包含在string 里面. 同理 any 应该也有类似的问题, 而且更严重

这个是 4.x 的文档？

**Comment by pearmini** [comment]
@oushu1suzhijie1  这里是指 5.x 还是 4.x 的文档？

**Comment by wilson921108** [comment]
> > any还是没用到过, 不过string有碰到, 比如
> > export declare type ColorAttr = string | string[] | ((datum: Datum, defaultColor?: string) => string);
> > 实际使用的时候其实除了 string 还有 另一种声明颜色的方式 "l(90) 0:#ffffff 1:green 0.15" 这种算是渐变色. 但是被包含在string 里面. 同理 any 应该也有类似的问题, 而且更严重
> 
> 这个是 4.x 的文档？

我目前用的G2-plot : 2.4.20
依赖是G2  4.2.9

顺便问下,V5 的文档地址

**Comment by oushu1suzhijie1** [comment]
> @oushu1suzhijie1 这里是指 5.x 还是 4.x 的文档？

5.X encode用法以及好多api的用法在api文档中根本找不到

**Comment by pearmini** [comment]
> > @oushu1suzhijie1 这里是指 5.x 还是 4.x 的文档？
> 
> 5.X encode用法以及好多api的用法在api文档中根本找不到

嗯嗯，目前是写在教程里面的，正在做迁移工作，encode 可以在这里看：https://g2.antv.antgroup.com/manual/single-view/mark#%E9%80%9A%E9%81%93

**Comment by pearmini** [comment]
@ wilson921108 v5 文档地址：https://g2.antv.antgroup.com/manual/introduction

**Comment by hustcc** [comment]
related: https://github.com/antvis/G2/discussions/4973
